snooty/parser.py

```diff
@@ -424,6 +438,60 @@ def dispatch_departure(self, node: docutils.nodes.Node) -> None:
                 target.children = [identifier]
                 item.term.append(target)
 
+    def handle_tabset(self, node: n.Directive, line: int) -> None:
+        tabset = node.options["tabset"]
+        # retrieve dictionary associated with this specific tabset
+        if tabset not in specparser.SPEC.tabs:
+            self.diagnostics.append(UnknownTabset(tabset, line))
+            print("unknown tabset!! ")
+            return
+
+        tab_definitions_list = specparser.SPEC.tabs[tabset]
+        tabid_list: List[str] = []
+
+        for idx, child in enumerate(node.children):
+            if not isinstance(child, n.Directive) or child.name != "tab":
+                self.diagnostics.append(
+                    TabMustBeDirective(str(type(child).__class__.__name__), line)
+                )
+                return
+
+            tabid = child.options["tabid"]
```

Comment:
What if there's no tabid option?

You can do `tabid = child.options.get("tabid")` I think, and make the below line `if not tabid or not isinstance...`

Reply:
Oh I see, you checked in the caller